toolchain/check/testdata/class/no_prelude/destroy.carbon

```diff
@@ -8,142 +8,353 @@
 // TIP: To dump output, run:
 // TIP:   bazel run //toolchain/testing:file_test -- --dump_output --file_tests=toolchain/check/testdata/class/no_prelude/destroy.carbon
 
-// --- fail_todo_basic.carbon
+// --- self.carbon
 
 library "[[@TEST_NAME]]";
 
 class C {
-  // CHECK:STDERR: fail_todo_basic.carbon:[[@LINE+4]]:6: error: semantics TODO: `KeywordNameBeforeParamsId` [SemanticsTodo]
-  // CHECK:STDERR:   fn destroy[self: Self]();
-  // CHECK:STDERR:      ^~~~~~~
-  // CHECK:STDERR:
   fn destroy[self: Self]();
 }
 
-// --- fail_todo_addr.carbon
+// --- addr_self.carbon
 
 library "[[@TEST_NAME]]";
 
 class C {
-  // CHECK:STDERR: fail_todo_addr.carbon:[[@LINE+4]]:6: error: semantics TODO: `KeywordNameBeforeParamsId` [SemanticsTodo]
-  // CHECK:STDERR:   fn destroy[addr self: Self*]();
-  // CHECK:STDERR:      ^~~~~~~
-  // CHECK:STDERR:
   fn destroy[addr self: Self*]();
 }
 
-// --- fail_todo_class_function.carbon
+// --- explicit_return.carbon
+
+library "[[@TEST_NAME]]";
+
+class C {
+  fn destroy[self: Self]() -> ();
+}
+
+// --- fail_class_function.carbon
 
 library "[[@TEST_NAME]]";
 
 class C {
-  // CHECK:STDERR: fail_todo_class_function.carbon:[[@LINE+4]]:6: error: semantics TODO: `KeywordNameBeforeParamsId` [SemanticsTodo]
+  // CHECK:STDERR: fail_class_function.carbon:[[@LINE+4]]:3: error: missing implicit `self` parameter [DestroyFunctionMissingSelf]
   // CHECK:STDERR:   fn destroy();
-  // CHECK:STDERR:      ^~~~~~~
+  // CHECK:STDERR:   ^~~~~~~~~~~~~
   // CHECK:STDERR:
   fn destroy();
 }
 
-// --- fail_todo_destroy_in_namespace.carbon
+// --- fail_extra_implicit_params.carbon
 
 library "[[@TEST_NAME]]";
 
-namespace NS;
+class C {
+  // CHECK:STDERR: fail_extra_implicit_params.carbon:[[@LINE+4]]:26: error: unexpected implicit parameter [DestroyFunctionUnexpectedImplicitParam]
+  // CHECK:STDERR:   fn destroy[self: Self, T:! type]();
+  // CHECK:STDERR:                          ^
+  // CHECK:STDERR:
+  fn destroy[self: Self, T:! type]();
```

Comment:
Maybe also a testcase where T comes first? I see the code handles it and diagnoses the correct position